### PR TITLE
fixes bug where role update is reflected as notes update

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/16950

this issue stems from the ``moped_proj_personnel_roles`` table not being associated with a ``project_id`` (rendering that field as ``null``) and therefore not returning records as part of the ``moped_activity_log`` query. investigating this has uncovered some potential tech debt when it comes to the activity log triggers (as noted in this pr), but @mddilley and i decided in the short term to fix the activity log rendering incorrect associations in the ui by assuming that a change made to the ``moped_proj_personnel`` table that is not related to a user or note is related to a role.

## Testing
**URL to test:** 

https://deploy-preview-1424--atd-moped-main.netlify.app/

**Steps to test:**
- go to a project team and add or edit a team member
- change their role, the change should be reflected in the activity log
- change their notes, the change should also be reflected in the activity log

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/utils/activityLogFormatters/mopedPersonnelActivity.js [hunk 1]
```diff
@@ -41,12 +41,24 @@ export const formatPersonnelActivity = (change, userList) => {
...
+  if (change.description[0].field === "notes")
+    return {
     changeIcon,
     changeText: [
       { text: "Updated team member notes for " },
       { text: userList[changeData.new.user_id], style: "boldText" },
     ],
   };
+
+  // the only other change that would be reflected here is adding or removing roles
```
Comment (mddilley) [inline moped-editor/src/utils/activityLogFormatters/mopedPersonnelActivity.js:54]: thanks for adding the extra condition for notes updates that we can check by the field changed. 

I'm keeping in mind that this catch-all for detecting role change could create a future bug if we ever add a new field to personnel records, but I think that chance is low.

**Review by mddilley — CHANGES_REQUESTED** [review]
Thanks for patching this one up! 🙌 One small change is needed before merge. 🙏

File: moped-database/metadata/tables.yaml [hunk 2]
```diff
@@ -3439,6 +3439,9 @@
       request_transform:
         body:
           action: transform
+          // this transformation or the schema itself may need to be refactored
+          // as it does not provide a project id or personnel id to associate this
+          // table with moped_proj_personnel in the activity log
```
[the remark this answers is not in the document]
Comment (tillyw) [reply moped-database/metadata/tables.yaml]: yes sorry about that! definitely not the place for those comments

File: moped-editor/src/utils/activityLogFormatters/mopedPersonnelActivity.js (hunk 1 above)
Comment (tillyw) [reply moped-editor/src/utils/activityLogFormatters/mopedPersonnelActivity.js:54]: thanks!

File: moped-database/metadata/tables.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mddilley) [reply moped-database/metadata/tables.yaml]: thanks!

**Review by mddilley — APPROVED** [review]
I tested updated both personnel roles and notes. I see the activity feed showing accurate descriptions of those changes. I also tried replacing a team member and removing one, and all of those entries look the same as before. Thanks for fixing! 🙌 🚀 🚢

**Review by chiaberry — APPROVED** [review]
Thanks for fixing!

**Pull request merged**